vdr/doc/resolvers.go

```diff
@@ -61,7 +61,7 @@ func (d Resolver) resolve(id did.DID, metadata *types.ResolveMetadata, depth int
 		return nil, nil, err
 	}
 
-	if metadata != nil && !metadata.AllowDeactivated {
+	if len(doc.Controller) > 0 && (metadata == nil || !metadata.AllowDeactivated) {
```

Comment:
shouldn't this be:

```
if len(doc.Controller) > 0 && (metadata == nil && !metadata.AllowDeactivated)
```

?

Because now you can't resolve the DID document with `AllowDeactivated == true` when the document has controllers, at all?

Reply:
`(metadata == nil && !metadata.AllowDeactivated)` will resolve in a nullpointer exception.

Why would it not be possible to resolve the docuent when `allowDeactivated == true`? at line 59 the document got resolved, regardless of the `allowDeactivated` option.